app/src/actioncommands.cpp

```diff
@@ -548,13 +548,29 @@ void ActionCommands::ZoomOut()
 void ActionCommands::rotateClockwise()
 {
     float currentRotation = mEditor->view()->rotation();
-    mEditor->view()->rotate(currentRotation + 15.f);
+    // Invert rotation direction if view is flipped either vertically or horizontally
+    if (mEditor->view()->isFlipHorizontal() ^ mEditor->view()->isFlipVertical())
```

Comment:
Hi Henrique

Thanks again for contributing 😄 

I have a few things this time around, although the code itself is _fine_, I think we can do a better than simply copy/pasting, which could introduce bugs down the line.

Bitwise XOR is not something we use in the project afaik. I don't personally have a stance on that as I think there's something to be said about its simplicity but it could introduce bugs... as such I think it may be better to use conventional logical operators instead, like:
if (A && !B || B && !A)

Reply:
That seems even less clear to me. I don't mind keeping the binary operator, but if we have to go with logical operators, `(A || B) && !(A && B)` seems a bit better to me.

Reply:
Both seem pretty unreadable to me personally  😅

@MrStevns Could you please elaborate on "it could introduce bugs"? I don't understand how a XOR operation could introduce bugs, specially since I thoroughly tested the behavior.

Reply:
also.. using operators like that would mean having to write "A" and "B" twice each, forcing me to declare them as constants which will make me have to choose between good naming and readability:

```C++
const bool h = mEditor->view()->isFlipHorizontal();
const bool v = mEditor->view()->isFlipVertical();

if (h && !v || v && !h)
// or
if ((h || v) && !(h && v))
```
vs
```C++
const bool flipHorizontal = mEditor->view()->isFlipHorizontal();
const bool flipVertical = mEditor->view()->isFlipVertical();

if (flipHorizontal && !flipVertical || flipVertical && !flipHorizontal)
// or
if ((flipHorizontal || flipVertical) && !(flipHorizontal && flipVertical))
```

To me personally it seems cluttered either way.

I found this one on [StackOverflow](https://stackoverflow.com/a/39473219) which seems pretty simple:
```C++
if (mEditor->view()->isFlipHorizontal() == !mEditor->view()->isFlipVertical())
```

Reply:
> @MrStevns Could you please elaborate on "it could introduce bugs"? I don't understand how a XOR operation could introduce bugs, specially since I thoroughly tested the behavior.

Sure, though I didn't mean to imply that the code would break on its own, just that bitwise comparison can become brittle if the implementation changes later in a way that the compiler does not complain about and the person updating the code is not aware. 

In reality that may never happen, I just think it's better to be safe than smart in that regard. 😅 

If the others are fine with bitwise xor comparison, then let's go with that, otherwise what you proposed is also fine:
> if (mEditor->view()->isFlipHorizontal() == !mEditor->view()->isFlipVertical())